sub-packages/bionemo-testing/src/bionemo/testing/testing_callbacks.py

```diff
@@ -14,171 +14,239 @@
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Union
+
+import numpy as np
+import torch
+from nemo.lightning import io
+from nemo.lightning.data import MegatronPretrainingSampler
+from nemo.lightning.megatron_parallel import CallbackMethods, DataT, MegatronLossReduction, MegatronStep
+from overrides import override
 from pytorch_lightning import Callback, LightningModule, Trainer
-from torch.nn import functional as F
-from torch.utils.data import DataLoader
-
-
-def compute_biobert_loss_singlegpu(model, dl: DataLoader):
-    """Computes the loss for BioBert models on a single GPU.
-
-    This will not function in multi-gpu settings nor with models that do not conform to BioBert.
-
-    Args:
-        model (torch.nn.Module): The Biobert model.
-        dl (torch.utils.data.DataLoader): The data loader.
 
-    Returns:
-        float: The mean loss.
-
-    See Also:
-    - :class: BioBertModel
-    """
-    n, loss = 0, 0.0
-    model.eval()
-    # batch = next(iter(dl))
-    batch = model.data_step(iter(dl))
-    result = model(
-        input_ids=batch["text"].cuda(),  # 'tokens' also a valid input for MockGPTDataModule
-        attention_mask=batch["attention_mask"].cuda(),
-    )
-    loss_mask = batch["loss_mask"].cuda()
-    # Not guaranteed i guess?
-    logits = result["token_logits"]
-    target = batch["labels"].cuda()
-    loss += F.cross_entropy(logits[loss_mask].float(), target[loss_mask], reduction="sum")
-    n += loss_mask.sum()
-
-    mean_loss: float = (loss / n).detach().cpu().numpy().item()
-    model.train()
-    return mean_loss
+from bionemo.testing.harnesses.mode import Mode
+from bionemo.testing.torch import recursive_detach
 
 
 class StopAndGoException(Exception):  # noqa: D101
     pass
 
 
 class RaiseAfterMetadataCallback(Callback):
-    """A callback that raises a StopAndGoException kills it if the metadata from the MetadataSaveCallback was saved successfully beforehand.
+    """A callback that raises a StopAndGoException after the validation epoch.
 
     Use this callback for pytest based Stop and go tests.
     """
 
-    def __init__(self, metadata_path: pathlib.Path):  # noqa: D107
-        self.metadata_path = metadata_path
+    def on_validation_epoch_end(self, trainer: Trainer, pl_module: LightningModule):  # noqa: D102
+        if trainer.sanity_checking:
+            return
+        raise StopAndGoException()
 
-    def on_train_batch_start(self, trainer: Trainer, pl_module: LightningModule, batch: Any, batch_idx: int):
-        """PTL callback that raises a StopAndGoException if metadata exists."""
-        pickle_file_path = os.path.join(self.metadata_path, "checkpoints/metadata.pkl")
-        if os.path.exists(pickle_file_path):
-            # Register the signal handler
-            raise StopAndGoException("Terminating early, checkpoint exists.")
-            # kill job afterwards
 
+class BaseInterruptedVsContinuousCallback(Callback, CallbackMethods, io.IOMixin):
```

Comment:
Nice that we now have this serialized as well.

Reply:
Can we add more docstring on `CallbackMethods`? I doubt new users will know what it is.

Reply:
Sure, I added a mention that this is where you can find the available megatron callback functions.